### DISCUSSION TITLE
feat: support System.Text.Json

### DISCUSSION THREAD

**Discussion by tsanton** [Gremlinq/ExRam.Gremlinq#1561] [Ideas]
**Is your feature request related to a problem? Please describe.**
I would like to be able to to use System.Text.Json with Gremlinq

**Describe the solution you'd like**
Create a support package for System.Test.Json along the line of `ExRam.Gremlinq.Support.NewtonsoftJson`.

**Describe alternatives you've considered**
Sticking to NewtonsoftJson, but my company has fully migrated to System.Text.Json and does not want to use two json libraries.

**Additional context**
System.Text.Json is becoming omnipresent and a preferred choice for many. Implementing support would make adoption of this library easier for many.

**Consider sponsoring me**
Depending on timeline and amount of work we can of course contribute in order to have it in place :)

**Comment by danielcweber** [comment]
It's in the works, although as a commercial offering. Stay tuned.

**Comment by tsanton** [comment]
@danielcweber whilst I'm at it (tell me to create a new issue if thats preferred), but is there any simple way to AddRangeAsync? I know you can AddV().AddV() but I have the need batch create dynamic amounts of vertexes and I'd love to save the roundtrips.

Further, is it possible to AddV with multiple Edges? as pr the docs this is possible:

```
await _g
        .AddV(new Person { Name = "Bob", Age = 36 })
        .AddE<Knows>()
        .To(__ => __
            .AddV(new Person { Name = "Jeff", Age = 27 }))
```

But I'm not able to make this work (or with more):

```
await _g
        .AddV(new Person { Name = "Bob", Age = 36 })
        .AddE<Knows>()
        .To(__ => __
            .AddV(new Person { Name = "Jeff", Age = 27 }))
        .AddE<Knows>()
        .To(__ => __
           .V(AnotherPerson.Id))
```

Lastly there's the need to batch created vertexes with one or more OutE..

Any pointers to docs would be greatly appreciated! If doable, but impractical to explain, please let me know and I can book a session :)

**Comment by danielcweber** [comment]
This feature is now available as extension library!

https://docs.gremlinq.net/extensions/
